File: 30-Day LeetCoding Challenge/C++/maxProfit.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
    public:
        int maxProfit(vector<int>& prices) {
            int max_profit = 0;
            for(int i = 1; i < prices.size(); i++) {
                if (prices[i-1] < prices[i]) {
                    max_profit += prices[i] - prices[i-1];
                }
            }
            return max_profit;
        }
};

int main() {
    Solution s;
    vector<int> prices{1,6,2,3,1};
    cout << s.maxProfit(prices);
    return 0;
}

File: 30-Day LeetCoding Challenge/Python/maxProfit.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        max_profit = 0
        for i in range(1,len(prices)):
            if prices[i-1] < prices[i]:
                max_profit += prices[i] - prices[i-1]
        return max_profit
s = Solution()
nums = [5,1,6,2,1,7]
s.moveZeroes(nums)
print(nums)